web/src/analytics.ts

```diff
@@ -0,0 +1,41 @@
+import { useConnectionState } from "@/data/hooks";
+import { setRtdmLocalStorage, getRtdmLocalStorage } from "@/local-storage";
+import { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { v4 as uuidv4 } from "uuid";
+
+export function trackAnalyticsEvent(
+  event: string,
+  params?: Record<string, string | number>
+) {
+  window.analytics.track(`rtdm-${event}`, params);
+}
+
+export function identifyUser() {
+  let userID = getRtdmLocalStorage().userID;
+  if (!userID) {
+    userID = uuidv4();
+    setRtdmLocalStorage({ userID });
+  }
+  window.analytics.identify(userID);
```

Comment:
just for additional safety, wrap these calls with a `if (window.analytics)` check and if it's not defined just log with `console.warn` but otherwise don't throw an exception.